controllers/project_mesh_controller.go

```diff
@@ -71,17 +71,7 @@ func (r *OpenshiftServiceMeshReconciler) SetupWithManager(mgr ctrl.Manager) erro
 }
 
 func meshAwareNamespaces(object client.Object) bool {
-	if IsReservedNamespace(object.GetName()) {
-		return false
-	}
-
-	for annotation := range object.GetAnnotations() {
-		if annotation == AnnotationServiceMesh {
-			return true
-		}
-	}
-
-	return false
+	return !IsReservedNamespace(object.GetName()) && object.GetAnnotations()[AnnotationServiceMesh] != ""
```

Comment:
what about "false" ;)

Reply:
It's handled in the reconcile

Reply:
Gonna rework these predicates to also handle the case when the annotation has been removed.

Reply:
Done in [df13e8d](https://github.com/maistra/odh-project-controller/pull/89/commits/df13e8d7ff29a703a5d6193570172a47cac04095)